### PR TITLE
Update cache link logic.

### PR DESCRIPTION
**Pull request by reitermarkus**
Depends on https://github.com/Homebrew/brew/pull/4662.

### CODE REVIEW

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -775,7 +775,15 @@ def bottle_reinstall_formula(formula, new_formula)
...
 
-      FileUtils.ln bottle_filename, HOMEBREW_CACHE/bottle_filename, force: true
+      bottle_json = JSON.parse(File.read(bottle_json_filename))
+
+      root_url = bottle_json.dig(formula.name, "bottle", "root_url")
+      filename = bottle_json.dig(formula.name, "bottle", "tags").values.first["filename"]
+      url_sha256 = Digest::SHA256.hexdigest("#{root_url}/#{filename}")
+
+      FileUtils.ln bottle_filename, HOMEBREW_CACHE/"downloads/#{url_sha256}--#{filename}", force: true
+      FileUtils.ln_s "downloads/#{url_sha256}--#{filename}", HOMEBREW_CACHE/bottle_filename, force: true
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: Could this maybe get refactored into some method (doesn't have to be public) in Homebrew/brew that test-bot can call instead? Don't really like the logic being duplicated here.

Comment (reitermarkus) [reply cmd/brew-test-bot.rb]: Any suggestion on where to put it? I thought about it but I can't think of a good place where this logic should go. Even in the download strategies, these steps don't happen in a single cohesive action, and `FileUtils.ln` is specific to `test-bot`.

**Comment by reitermarkus** [comment]
@MikeMcQuaid, I changed it to used the `CurlDownloadStrategy` strategy to get the proper name.

Instead of the `FileUtils.ln_s` step, we could now also use `download_strategy.fetch`, which will show “Already downloaded” and create the symlink.

**Comment by MikeMcQuaid** [comment]
> Instead of the FileUtils.ln_s step, we could now also use download_strategy.fetch, which will show “Already downloaded” and create the symlink.

This works 👍

**Pull request merged**